AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java

```diff
@@ -291,13 +293,31 @@ private AndroidManifest parse(File androidManifestFile, boolean libraryProject)
 		componentQualifiedNames.addAll(receiverQualifiedNames);
 		componentQualifiedNames.addAll(providerQualifiedNames);
 
+		Map<String, String> metaDataQualifiedNames = new HashMap<String, String>();
+		NodeList metaDataNodes = documentElement.getElementsByTagName("meta-data");
+		for (int i = 0; i < metaDataNodes.getLength(); i++) {
+			Node node = metaDataNodes.item(i);
+			Node nameAttribute = node.getAttributes().getNamedItem("android:name");
+			Node valueAttribute = node.getAttributes().getNamedItem("android:value");
+
+			if (nameAttribute == null || valueAttribute == null) {
```

Comment:
OOps, i missed the already existing `if`s. In this case, i think you did the right thing, this way the code is consistent. We can refactor all `if`s later.